src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -158,31 +161,42 @@ export class PatientViewPageStore
             }
     }}, []);
 
-    readonly mrnaExprRankData = remoteData<MrnaRankData>({
+    @action("ChangePatientId") changePatientId(newId: string) {
+        this.patientId = newId;
+    }
+    readonly requestMrnaExprRankData = remoteData({
         await: () => [
-            this.mrnaRankGeneticProfileId,
-            this.mutationData
+            this.mrnaRankGeneticProfileId
         ],
         invoke: () => {
-            const sampleToEntrezGeneIds = _.reduce(this.mutationData.result, (map:{ [s:string]:Set<number> }, next:Mutation) => {
-                const sampleId = next.sampleId;
-                map[sampleId] = map[sampleId] || new Set();
-                map[sampleId].add(next.entrezGeneId);
-                return map;
-            }, {});
-
-            return new Promise((resolve, reject) => {
+            return Promise.resolve(function(sampleToEntrezGeneIds:{ [sampleId:string]:Set<number> }) {
+                // Reference to new immutable map
+                // See which we need to fetch, and set "pending" for those data
+                const toQuery:{ [sampleId:string]:number[]} = {};
+                for (const sampleId in sampleToEntrezGeneIds) {
+                    if (sampleToEntrezGeneIds.hasOwnProperty(sampleId)) {

```

Comment:
this check should really be handled by iterator.  so, for example, use Object.keys or lodash _.forEach.